packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -242,17 +243,30 @@ abstract class BoxBorder extends ShapeBorder {
     }
   }
 
-  static void _paintNonUniformBorder(
+  /// Paints a Border with different widths, styles and strokeAligns, on any
+  /// borderRadius while using a single color.
+  ///
+  /// See also:
+  ///
+  ///  * [paintBorder], which supports multiple colors but not borderRadius.
+  ///  * [paint], which calls this method.
+  static void paintNonUniformBorder(
     Canvas canvas,
     Rect rect, {
     required BorderRadius? borderRadius,
-    required BoxShape shape,
     required TextDirection? textDirection,
-    required BorderSide left,
-    required BorderSide top,
-    required BorderSide right,
-    required BorderSide bottom,
+    BoxShape shape = BoxShape.rectangle,
+    BorderSide? top,
+    BorderSide? right,
+    BorderSide? bottom,
+    BorderSide? left,
+    required Color color,
   }) {
+    top ??= BorderSide.none;
+    right ??= BorderSide.none;
+    bottom ??= BorderSide.none;
+    left ??= BorderSide.none;
```

Comment:
these can be default values for the parameters

Reply:
I don't think it can, because I want them to be null.

Reply:
why do you want them to be nullable?

Reply:
I guess I thought the code would be cleaner/easier. Making them not null at this moment..